onnxruntime/core/providers/webnn/builders/impl/gru_op_builder.cc

```diff
@@ -43,18 +45,22 @@ Status GruOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const No
   const auto& input_defs = node.InputDefs();
   std::vector<int64_t> input_shape;
   ORT_RETURN_IF_NOT(GetShape(*input_defs[0], input_shape, logger), "Cannot get input's shape");
-  uint32_t steps = static_cast<uint32_t>(input_shape[0]);
-
+  uint32_t steps = 1;
+  if (!input_shape.empty()) {
```

Comment:
Shouldn't it be an error in that case, rather than setting steps = 1? From the ONNX spec, input must be:

> The input sequences packed (and potentially padded) into one 3-D tensor with the shape of [seq_length, batch_size, input_size].

The `input_shape` should never be empty - I just wanted to avoid missing a validation check in case one snuck through from a malformed model.

Reply:
Got it and restored the code.